helm/hpcc/templates/_helpers.tpl

```diff
@@ -491,21 +491,27 @@ vaults:
 {{- end -}}
 
 {{/*
-Return a value indicating whether a storage plane is defined or not.
+Check whether a storage plane is defined or not.
 */}}
 {{- define "hpcc.isValidStoragePlane" -}}
 {{- $search := .search -}}
+{{- $category := .category -}}
 {{- $storage := (.root.Values.storage | default dict) -}}
 {{- $planes := ($storage.planes | default list) -}}
 {{- $dataStorage := ($storage.dataStorage | default dict) -}}
 {{- /* If storage.dataStorage.plane is defined, the implicit plane hpcc-dataplane is not defined */ -}}
-{{- $done := dict "matched" (and (not $dataStorage.plane) (eq $search "hpcc-dataplane")) -}}
+{{- $done := dict "matched" (and (not $dataStorage.plane) (eq $search "hpcc-dataplane")) "all" "" -}}
 {{- range $plane := $planes -}}
- {{- if eq $search $plane.name -}}
- {{- $_ := set $done "matched" true -}}
+ {{- if has $category $plane.labels -}}
```

Comment:
better to reverse the tests, and report an error if the category isn't in the labels.  Also need to be careful at the moment that a blank labels list is treated as equivalent to [data]

Reply:
Except I'm scanning through all planes with this category in order to give the candidate list in the error message